modules/explorer.go

```diff
@@ -98,14 +67,29 @@ type (
 		// provided unlock hash.
 		UnlockHash(types.UnlockHash) []types.TransactionID
 
+		// SiacoinOutput will return the siacoin output associated with the
+		// input id.
+		SiacoinOutput(types.SiacoinOutputID) (types.SiacoinOutput, bool)
+
 		// SiacoinOutputID returns all of the transaction ids associated with
 		// the provided siacoin output id.
 		SiacoinOutputID(types.SiacoinOutputID) []types.TransactionID
 
+		// FileContractHistory returns the history associated with a file
+		// contract.  History is broken in to 3 elements, and a bool is
+		// returned indicating whether history exists as a part of each
+		// element. If the file contract itself does not exist, no other type
+		// of history will exist either.
+		FileContractHistory(types.FileContractID) (types.FileContract, []types.FileContractRevision, types.StorageProof, bool, bool, bool)
```

Comment:
this is a bit messy...consider splitting into multiple functions or combining return values into a struct. Three "exists" bools is just too much.